src/reporter/reporter.py

```diff
@@ -147,7 +157,15 @@ def notify():
         fiware_sp = request.headers.get('fiware-servicepath', None)
     else:
         fiware_sp = None
-
+    res_entity = []
+    e = None
+    for entity in payload:
+        # Validate entity update
+        e = _filter_empty_entities(entity)
+        if e is not None:
+            res_entity.append(e)
```

Comment:
If you could move this logic up at as I suggested in my other comment, it'd be great!